packages/apollo-server-codegen/src/Translators/TypeScriptTranslator.ts

```diff
@@ -0,0 +1,364 @@
+import { Translator } from ".";
+import * as IR from "../IR";
+
+export class TypeScriptTranslator extends Translator {
+  public generate(
+    objects: IR.ObjectDefinition[],
+    enums: IR.EnumDefinition[],
+    scalars: IR.ScalarDefinition[],
+    unions: IR.UnionDefinition[],
+    inputObjects: IR.InputObjectDefinition[]
+  ): string {
+    const header = this.generateHeader();
+    const resolvers = this.generateTopLevelResolvers(
+      objects.map(d => d.name),
+      enums.map(e => e.name),
+      scalars.map(s => s.name)
+    );
+
+    return [
+      header,
+      resolvers,
+      ...[...objects, ...enums, ...scalars, ...unions, ...inputObjects].map(
+        def => def.translate(this)
+      )
+    ].join("\n");
+  }
+
+  private generateHeader() {
+    // Generate some utility higher order types that we'll reference later on.
+    return `// This is a machine generated file.
+// Use "apollo service:codegen" to regenerate.
```

Comment:
can we avoid printing these in every file? Will this pollute other types?

Reply:
There is only one generated file, and no, type definitions in a module won't impact other modules without first explicitly importing them in that module.